core/rint/test/TRintTests.cxx

```diff
@@ -0,0 +1,41 @@
+#include <iostream>
+#include <string>
+
+#include "gtest/gtest.h"
+
+#include "TRint.h"
+
+TEST(TRint, UnrecognizedOptions)
+{
+   // Create array of options
+   int argc{4};
+   char **argv = new char *[argc];
+   argv[0] = const_cast<char *>("-q");
+   argv[1] = const_cast<char *>("-z");
+   argv[2] = const_cast<char *>("--nonexistingoption");
+   argv[3] = const_cast<char *>("-b");
```

Comment:
might simplify this a lot if it works:

```cpp
std::vector<const char *> v{"-q", "-z", "--nonexistingoption", "-b"};
```

(and the clean-up logic at the end can go away)

Reply:
The problem is the [TRint constructor](https://github.com/root-project/root/blob/e553d2eba5146f89988b1ab80e97f78a72e2b004/core/rint/src/TRint.cxx#L144) that accepts only a `char**` and the warnings that are generated if one tries to [initialize a `char **`](https://godbolt.org/z/GrT354vjr): ISO C++ forbids converting a string constant to 'char*'

Reply:
you can `const_cast` `v.data()`, I think?

Reply:
Yes that works, I'm unsure as how that will work with the fact that `TRint` will modify the array (eliminating the recognized options and leaving only unrecognized ones). I'll build with the modification and let you know

Reply:
oh, no that doesn't work at all :sweat_smile: nevermind